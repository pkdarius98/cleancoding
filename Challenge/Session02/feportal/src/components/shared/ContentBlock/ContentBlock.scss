.content-block {
  border-radius: $border-radius-base;
  line-height: $line-height-20;

  .ant-card-head {
    padding: 0 $align-base;
    border-bottom: none;
    min-height: auto;

    .ant-card-head-wrapper {
      align-items: flex-start;

      .ant-card-head-title,
      .ant-card-extra {
        padding-bottom: 0;
      }

      .ant-card-head-title {
        font-size: $font-size-16;
        font-weight: $font-weight-bold;
      }
    }
  }

  .ant-card-body {
    padding: $align-base;

    .ant-table-pagination {
      margin-bottom: 0;
    }
  }

  &.has-divider {
    .ant-card-head {
      border-bottom: $border-base;

      .ant-card-head-title,
      .ant-card-extra {
        padding-bottom: $align-base;
      }
    }
  }

  &:last-child {
    margin-bottom: 0;
  }
}
