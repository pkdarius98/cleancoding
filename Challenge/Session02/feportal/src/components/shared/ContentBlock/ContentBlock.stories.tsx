import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import ContentBlock, { ContentBlockProps } from './ContentBlock';

export default {
  title: 'Components/ContentBlock',
  component: ContentBlock,
  parameters: {
    docs: {
      description: {
        component:
          'Wrap content of each block inside this component. You can add title, extra, divider between title and content... for each block usage.',
      },
    },
  },
} as Meta;

const Template: Story<ContentBlockProps> = args => (
  <div style={{ padding: 16, background: '#f5f5f5' }}>
    <ContentBlock {...args} />
  </div>
);

export const Default = Template.bind({});
Default.args = {
  title: 'Title',
  extra: (
    <a
      href="https://storybook.js.org/"
      target="_blank"
      rel="noopener noreferrer"
    >
      Extra link
    </a>
  ),
  children: 'This is Content Block',
};

export const HasDivider = Template.bind({});
HasDivider.args = {
  ...Default.args,
  hasDivider: true,
};
