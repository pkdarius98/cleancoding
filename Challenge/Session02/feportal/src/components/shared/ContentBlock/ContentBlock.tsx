import React from 'react';
import { Card, CardProps } from 'antd';
import classNames from 'classnames';
import './ContentBlock.scss';

export interface ContentBlockProps extends CardProps {
  /**
   * Has divider between title and body?
   */
  hasDivider?: boolean;
}

const ContentBlock: React.FC<ContentBlockProps> = props => {
  const { className, hasDivider, children, ...rest } = props;

  return (
    <Card
      className={classNames({
        'content-block': true,
        'has-divider': hasDivider,
        ...(className && { [className]: true }),
      })}
      bordered={false}
      {...rest}
    >
      {children}
    </Card>
  );
};

export default ContentBlock;
