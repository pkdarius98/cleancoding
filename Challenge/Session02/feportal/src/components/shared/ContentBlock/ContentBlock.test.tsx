import React from 'react';
import { render } from '@testing-library/react';
import ContentBlock from './ContentBlock';

it('Renders correctly', () => {
  const { container } = render(<ContentBlock />);
  expect(container.firstChild).toMatchSnapshot();
});
