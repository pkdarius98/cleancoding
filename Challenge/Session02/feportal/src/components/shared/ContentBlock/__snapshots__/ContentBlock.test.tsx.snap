// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  class="ant-card content-block"
>
  <div
    class="ant-card-body"
  />
</div>
`;
